sympy/vector/coordsysrect.py

```diff
@@ -1,19 +1,45 @@
+import types
+from sympy.simplify import simplify
 from sympy.core.basic import Basic
 from sympy.vector.scalar import BaseScalar
-from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol
+from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol, symbols, sqrt, Lambda, Dummy
 from sympy.core.compatibility import string_types, range
 from sympy.core.cache import cacheit
 from sympy.vector.orienters import (Orienter, AxisOrienter, BodyOrienter,
                                     SpaceOrienter, QuaternionOrienter)
-import sympy.vector
 
 
-class CoordSysCartesian(Basic):
+def build_instance_from_lambda(fun=lambda x, y, z: (x, y, z)):
```

Comment:
OK.  The metric simplification does take a long time.  When I did it by hand in iSympy it didn't take near as long.  I'll get going on the changes for the metric_lambda, depreciation, tests.

Should we keep CoordSys3D() while the class handles only dim(3,) or go ahead and change the name to CoordSystem()?